internal/filter/filter.go

```diff
@@ -127,6 +132,13 @@ func (gf *globFilter) Match(name string, tags map[string]string) bool {
 	return true
 }
 
+func (gf *globFilter) IsHistogramMetric(name string) bool {
+	if gf.metricsToHisList != nil && gf.metricsToHisList.Match(name) {
```

Comment:
We can save the Match results on a map `map[string]bool` and only execute the regex if it haven't been already executed for a metric...

Something like this:

```
if isHisto, found := isHistionMap[metricname]; !found {
    isHistio = match(...)
    isHistioMap[metricname] = isHistio
    return isHistio
} else {
    return isHistio
}
```